### PR TITLE
chore!: reorder `Interface.decodeLog` arguments

### PR DESCRIPTION
**Pull request by nedsalk**
# Summary

Having this order makes more sense.

# Breaking Changes

The `Interface.decodeLog` arguments got reordered so that it now first accepts the `logId` and then the data.
```ts
// before
const abiInterface = new Interface(yourAbi);
const decodedLog = abiInterface.decodeLog(logData, logId);
```
```ts
// after
const decodedLog = abiInterface.decodeLog(logId, logData);
```

# Checklist

- [ ] I **_added_** — `tests` to prove my changes
- [ ] I **_updated_** — all the necessary `docs`
- [x] I **_reviewed_** — the entire PR myself, using the GitHub UI
- [ ] I **_described_** — all breaking changes and the Migration Guide

### CODE REVIEW

**Review by petertonysmith94 — APPROVED** [review]

**Review by Torres-ssf — APPROVED** [review]

**Review by danielbate — COMMENTED** [review]
I'm bikeshedding here so this isn't a hill I'm willing to die on as you've got 2 approvals, but I don't think we should be making breaking changes just because something seems more correct. I don't think that's enough value to justify breaking consumer apps and them having to search or deduce a resolution. And we can rewrite the API in #1795. Thoughts?

**Comment by danielbate** [comment]
> I'm bikeshedding here so this isn't a hill I'm willing to die on as you've got 2 approvals, but I don't think we should be making breaking changes just because something seems more correct. I don't think that's enough value to justify breaking consumer apps and them having to search or deduce a resolution. And we can rewrite the API in #1795.

**Comment by arboleya** [comment]
@danielbate You're probably right. My guess is that the `typegen`/`coder` API will inevitably break in the next big push anyway. Focusing on the new ABI spec and generated files should do the job.

File: .changeset/curly-buttons-invent.md
Comment (arboleya) [inline .changeset/curly-buttons-invent.md:6]: Approving with a comment to hold auto-merge from kicking before conversations are settled.

**Review by arboleya — APPROVED** [review]

**Comment by danielbate** [comment]
@nedsalk #1795 is a `v1` change. I agree we can deliver value via the points you've mentioned and not introduce breaking changes. However, the current design has been proved and battle tested, and the real value is in rethinking and rewriting both the implementation and API.

I'm pro closing this PR. I don't think the breakage justifies the value it delivers.

**Comment by petertonysmith94** [comment]
I'd also favour closing this PR until v1 after reflection.

Thoughts @nedsalk?

**Comment by petertonysmith94** [comment]
There appears to consensus that we'll hold off on this change till:
- https://github.com/FuelLabs/fuels-ts/issues/1795

Closing until then.

**Pull request closed without merging**
